### ISSUE TITLE
MD072: 'rumdl' warns that 'key-order' is an unknown option

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#467]
### rumdl version

0.1.28

### Installation method

mise

### Bug description

MD072 includes a configuration option `key-order`: https://github.com/rvben/rumdl/blob/265cc07c002f5db6d92d95d41ca1f7de10a24146/docs/md072.md#configuration.

Even though this option works fine, `rumdl` warns that it's invalid.

### Steps to reproduce

1. Create a Markdown file with this content:
   ```markdown
   ---
   title: A heading
   description: A description
   ---

   Some text.
   ```
1. Create a `.rumdl` file with this content:
   ```toml
   [MD072]  # https://rumdl.dev/md072/
   enabled = true
   key-order = ["description", "title"]
   ```
1. Run `rumdl` on the Markdown file.

### Expected behavior

When running `rumdl check`, I expect this output:

```shell
test.md:2:1: [MD072] YAML frontmatter keys are not sorted alphabetically: 'description' should come before 'title' [*]
```

### Actual behavior

When running `rumdl check`, I get this output:

```shell
[config warning] Unknown option for rule MD072: key-order
test.md:2:1: [MD072] YAML frontmatter keys are not sorted alphabetically: 'description' should come before 'title' [*]
```

### Configuration

```toml
[MD072]  # https://rumdl.dev/md072/
enabled = true
key-order = ["description", "title"]
```

### Operating system

macOS

**Comment by rvben** [comment]
Fixed in v0.1.30. The config validation was dropping nullable (`Option<T>`) fields from the rule schema, causing valid keys like `key-order` to be flagged as unknown. The schema now correctly preserves these fields. Thanks for the clear repro steps, @eread!
